src/cryptography/hazmat/primitives/asymmetric/ec.py

```diff
@@ -252,6 +264,28 @@ def __init__(self, algorithm):
     algorithm = utils.read_only_property("_algorithm")
 
 
+@utils.register_interface(KeyDerivationFunction)
+class ECDHnoKDF(object):
```

Comment:
If someone really needs the raw output they can always write a NullKDF very
easily.

On Sat, May 16, 2015 at 11:38 AM, Alex Stapleton <notifications@github.com>
wrote:

> In src/cryptography/hazmat/primitives/asymmetric/ec.py
> <https://github.com/pyca/cryptography/pull/1882#discussion_r30462024>:
>
> > @@ -252,6 +264,28 @@ def __init__(self, algorithm):
> >      algorithm = utils.read_only_property("_algorithm")
> >
> >
> > +@utils.register_interface(KeyDerivationFunction)
> > +class ECDHnoKDF(object):
>
> Having spoken with @alex <https://github.com/alex> I think it might
> actually be better to just leave this functionality out entirely until we
> have a clear use-case for it. How do you feel about that @simo5
> <https://github.com/simo5> ? Are you aware of any cases where you really
> need access to the raw ECDH output?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pyca/cryptography/pull/1882/files#r30462024>.
>



-- 
"I disapprove of what you say, but I will defend to the death your right to
say it." -- Evelyn Beatrice Hall (summarizing Voltaire)
"The people's good is the highest law." -- Cicero
GPG Key fingerprint: 125F 5C67 DFE9 4084